example/src/examples/http_proxy_repository.dart

```diff
@@ -27,15 +27,19 @@ class HttpProxyRepository extends PackageRepository {
         baseUrl.resolve('/api/packages/${Uri.encodeComponent(package)}');
 
     http.Response response = await client.get(versionUrl);
-    var json = convert.json.decode(response.body);
-    var versions = json['versions'] as List<Map>;
-    if (versions != null) {
-      for (var item in versions) {
-        var pubspec = item['pubspec'];
-        var pubspecString = convert.json.encode(pubspec);
-        yield new PackageVersion(pubspec['name'] as String,
-            pubspec['version'] as String, pubspecString);
+    if (response.statusCode == 200) {
+      var json = convert.json.decode(response.body);
+      var versions = json['versions'] as List<Map>;
+      if (versions != null) {
+        for (var item in versions) {
+          var pubspec = item['pubspec'];
+          var pubspecString = convert.json.encode(pubspec);
+          yield new PackageVersion(pubspec['name'] as String,
+              pubspec['version'] as String, pubspecString);
+        }
       }
+    } else {
+      yield null;
```

Comment:
I don't think we need to `yield null`, just do something like:
```
if (response.statusCode != 200) {
  return;
}
```

I think that'll be fine... then we don't need to filter afterwards...

Reply:
Agreed!